echopype/tests/calibrate/test_calibrate.py

```diff
@@ -45,7 +45,63 @@ def test_compute_Sv_returns_water_level(ek60_path):
     assert 'water_level' in ds_Sv.data_vars
 
 
-def test_compute_Sv_ek60_echoview(ek60_path):
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    ("use_swap", "chunk_dict"),
+    [
+        (True, None),
+        (True, {"channel": 1, "ping_time": 15, "range_sample": 500}),
+        (False, None),
+        (False, {"channel": 1, "ping_time": 15, "range_sample": 500})
+    ]
+)
+def test_compute_Sv_ek60_chunks_use_swap(use_swap, chunk_dict, ek60_path):
+    """Test outputs of `compute_Sv` on EK60 when `use_swap` and `chunk_dict` are provided."""
+    # Grab test file
+    ek60_raw_path = str(
+        ek60_path.joinpath('DY1801_EK60-D20180211-T164025.raw')
+    )
+
+    # Convert file
+    echodata = ep.open_raw(ek60_raw_path, sonar_model='EK60')
+
+    # Calibrate to get Sv
+    Sv = ep.calibrate.compute_Sv(
+        echodata,
+        use_swap=use_swap,
+        chunk_dict=chunk_dict
+    )["Sv"]
+
+    if use_swap:
+        # Check that there are only 2 layers of the task graph
+        assert 2 == len(Sv.data.dask.layers)
+    elif not use_swap and chunk_dict:
+        # Check that there are many more than 2 layers of the task graph
+        assert 77 == len(Sv.data.dask.layers)
+    elif not use_swap and not chunk_dict:
+        # Check that Sv data is not a dask array and is instead a numpy array
+        assert isinstance(Sv.data, np.ndarray)
```

Comment:
A small suggestion and a question:
- suggestion: how about merging the 2nd and 3rd cases so that you have
   ```python
   if use_swap:
      ...
   else:
      if chunk_dict:
         ...
      else:
         ...
   ```
- question: could you explain what dask is doing when you have `use_swap=False` but with `chunk_dict`? I guess I am not fully understanding why `use_swap=True` would do the trick to reduce the number of graph layers, whereas if you only set a `chunk_dict` but not using swap, it will produce many more layers.

Reply:
> could you explain what dask is doing when you have `use_swap=False` but with chunk_dict

My thought for this case was for the user to have the option to chunk `cal_object` without actually doing any computing, since `use_swap=True` would force the computation. In the case where the user may want to lazy load an entire pipeline from open raw to compute NASC, but now that I think about it, this could be done more simply and more explicitly by having something like:

```
ed = ed.chunk({'ping_time': 1000})
ds_Sv = ep.calibrate.compute_Sv(ed, use_swap=False)
...
ds_NASC = ...
ds_NASC.to_zarr(...compute=True)
```

> `use_swap=True` alone would do the trick to reduce the number of graph layers

Yeah you're right. In the `ep.open_raw(..., use_swap=False)` case, the `cal_object` would be fully computed and the Zarr Store wouldn't have any computation pointed towards it and it would instead just store something fully in-memory into the Zarr Store. I think you're right and I shouldn't allow for this case since it does nothing helpful. I should instead only allow `use_swap=True` if and only if the `cal_object` is chunked. And for that, I think maybe I should restrict chunking to outside of this function, like the example I show above? What are your thoughts on that?

The other option would be to restrict using `chunk_dict` to the case where `use_swap=True` and ignore `chunk_dict` in the case where `use_swap=False`.